src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -268,6 +268,29 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
 
             if (Avx2.IsSupported)
             {
+                if ((((nuint)Unsafe.AsPointer(ref searchSpace) + (nuint)offset) & (nuint)(Vector256<byte>.Count - 1)) != 0)
```

Comment:
The object couldn't be moved after the AsPointer check?

Reply:
e.g. the issue is that `string.strlen` calls this with length `int.MaxValue`; if the reads are not aligned then reading the Vector can cross a page boundary into a non-allocated page; creating a segfault, even if the part of the Vector in the page contains the `null` terminator.

Reply:
The method is used by public API https://source.dot.net/#System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs,253f89d386c4359c,references

Reply:
Yes, but the public api uses don't suggest you can use a length larger than the memory allocated for the array/span. This method will not read passed the length passed in.

The issue is because strlen is looking for a null terminator in an unknown length of memory; without this change it can page fault into non-process memory; with this change it can't (unless the string doesn't have a null terminator)

Reply:
Specifically https://github.com/dotnet/coreclr/blob/e78994d9d8f2d55bf6b06fb4daaec316a020bcf5/src/System.Private.CoreLib/shared/System/String.cs#L658-L659

Reply:
Aah..

Reply:
Right, we had two options on how to implement strlen: Either reuse `IndexOf` and make sure that it handles the `int.MaxValue` buffer size as unknown buffer size, or create a separate implementation for it that would be almost identical to `IndexOf`. We have chosen the first one.